ui/app/pages/swaps/build-quote/build-quote.js

```diff
@@ -215,6 +223,13 @@ export default function BuildQuote({
     );
   };
 
+  const tokenLinkForChain = isMainnet
```

Comment:
It would be better to use `rpcPrefs.blockExplorerURl` here, just like for the text portions. This would break on testnets, which don't have a custom block explorer URI.

And actually, I guess this would break on any custom network without a custom block explorer as well :thinking: 

I guess the ideal logic would be:

* If there is a custom block explorer, use that
* Else if the network is among the Infura networks, use Etherscan
* Else hide the explorer link - there is nothing to link to.

Reply:
Good points. Also, now I'm realizing we may want to default to https://bscscan.com for the binance chain, if no  `rpcPrefs.blockExplorerURl` is provided. This is the recommended equivalent of etherscan for that chain. It could be added to `shared/constants/swaps.js`. I'll check with product people on this.

Reply:
@Gudahtt a731fc97d applies logic very similar to what you suggested above. First it checks if there is a custom block explorer url, then it checks if the current chain is explicitly supported in `shared/constants/swaps.js`, then if it is among the etherscan networks, it uses etherscan. The same logic is also applied in awaiting-swap.js

On the build-quote page I decided to hide the link if there is nothing to link to, as it fits properly with the intention of this fix and the overall feature of supporting a custom networks for swaps. Right now, this situation can only happen on a testnet, but I think it is better that we don't ship broken links.